<a href="https://colab.research.google.com/github/StephaniePC1/ThisIsWhatIDoNow/blob/master/Copy_of_GP3_Ebay.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Exploring Ebay Car Sales Data
For this project, we will be working with a dataset from eBay Kleinanzeigen, a classifieds section located in the German eBay site.

The dataset contains information on the used cars found in eBay Kleinanzeigen. Our goal is to clean the data and analyze the included used car listings.

In [272]:
import numpy as np
import pandas as pd

path = "https://github.com/ldtcooper/Exploring-Ebay-Car-Sales/blob/master/autos.csv?raw=true"
auto = pd.read_csv(path, encoding='Latin-1')
auto

dateCrawled  ...             lastSeen
0      2016-03-26 17:47:46  ...  2016-04-06 06:45:54
1      2016-04-04 13:38:56  ...  2016-04-06 14:45:08
2      2016-03-26 18:57:24  ...  2016-04-06 20:15:37
3      2016-03-12 16:58:10  ...  2016-03-15 03:16:28
4      2016-04-01 14:38:50  ...  2016-04-01 14:38:50
...                    ...  ...                  ...
49995  2016-03-27 14:38:19  ...  2016-04-01 13:47:40
49996  2016-03-28 10:50:25  ...  2016-04-02 14:18:02
49997  2016-04-02 14:44:48  ...  2016-04-04 11:47:27
49998  2016-03-08 19:25:42  ...  2016-04-05 16:45:07
49999  2016-03-14 00:42:12  ...  2016-04-06 21:18:48

[50000 rows x 20 columns]

### Data Dictionary Provided for Orignal autos.csv Data Set:

> Indented block



We have included a short discription for each column in the data set. 
* `dateCrawled` - When this ad was first crawled. All field-values are taken from this date.
* `name` - Name of the car.
* `seller` - Whether the seller is private or a dealer.
* `offerType` - The type of listing
* `price` - The price on the ad to sell the car.
* `abtest` - Whether the listing is included in an A/B test.
* `vehicleType` - The vehicle Type.
* `yearOfRegistration` - The year in which the car was first registered.
* `gearbox` - The transmission type.
* `powerPS` - The power of the car in PS.
* `model` - The car model name.
* `kilometer` - How many kilometers the car has driven.
* `monthOfRegistration` - The month in which the car was first registered.
* `fuelType` - What type of fuel the car uses.
* `brand` - The brand of the car.
* `notRepairedDamage` - If the car has a damage which is not yet repaired.
* `dateCreated` - The date on which the eBay listing was created.
* `nrOfPictures` - The number of pictures in the ad.
* `postalCode` - The postal code for the location of the vehicle.
* `lastSeenOnline` - When the crawler saw this ad last online.

In [273]:
auto.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

We will modify the column names by changing the names of the columns to snake_case and reassigning more descriptive titles to some columns.

In [274]:
auto.rename({'dateCrawled':'date_crawled', 
             'offerType':'offer_type', 
             'abtest':'ab_test',
             'vehicleType':'vehicle_type', 
             'yearOfRegistration':'reg_year',
             'powerPS':'power_ps',
             'monthOfRegistration':'reg_month',
             'fuelType':'fuel_type',
             'notRepairedDamage':'not_repaired_damage',
             'dateCreated':'date_created',
             'nrOfPictures':'no_of_pictures',
             'postalCode':'postal_code',
             'lastSeen':'last_seen'}, axis=1, inplace=True)

In [275]:
auto.head()

date_crawled  ...            last_seen
0  2016-03-26 17:47:46  ...  2016-04-06 06:45:54
1  2016-04-04 13:38:56  ...  2016-04-06 14:45:08
2  2016-03-26 18:57:24  ...  2016-04-06 20:15:37
3  2016-03-12 16:58:10  ...  2016-03-15 03:16:28
4  2016-04-01 14:38:50  ...  2016-04-01 14:38:50

[5 rows x 20 columns]

In [276]:
auto.describe(include='all')

date_crawled         name  ...   postal_code            last_seen
count                 50000        50000  ...  50000.000000                50000
unique                48213        38754  ...           NaN                39481
top     2016-03-16 21:50:53  Ford_Fiesta  ...           NaN  2016-04-07 06:17:27
freq                      3           78  ...           NaN                    8
mean                    NaN          NaN  ...  50813.627300                  NaN
std                     NaN          NaN  ...  25779.747957                  NaN
min                     NaN          NaN  ...   1067.000000                  NaN
25%                     NaN          NaN  ...  30451.000000                  NaN
50%                     NaN          NaN  ...  49577.000000                  NaN
75%                     NaN          NaN  ...  71540.000000                  NaN
max                     NaN          NaN  ...  99998.000000                  NaN

[11 rows x 20 columns]

**Cleaning the data:**

The columns for `price` and `odometer` are used to store numeric values, but have extra characters and are currently being stored as text. We will now convert these columns to store the values as intergers. 

In [277]:
auto['price']=auto['price'].str.replace('$','').str.replace(',','').astype(float)

auto['price'].head()

0    5000.0
1    8500.0
2    8990.0
3    4350.0
4    1350.0
Name: price, dtype: float64

In [278]:
auto['odometer']=auto['odometer'].str.replace('km','').str.replace(',','').astype(float)

auto['odometer'].head()

0    150000.0
1    150000.0
2     70000.0
3     70000.0
4    150000.0
Name: odometer, dtype: float64

##Renaming the updated columns

In [279]:
auto.rename({'price':'price_$', 'odometer':'odometer_km'},axis=1,inplace=True)
auto.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price_$', 'ab_test',
       'vehicle_type', 'reg_year', 'gearbox', 'power_ps', 'model',
       'odometer_km', 'reg_month', 'fuel_type', 'brand', 'not_repaired_damage',
       'date_created', 'no_of_pictures', 'postal_code', 'last_seen'],
      dtype='object')

### Observations 
* There are two text columns where almost all of the values are the same: `seller` and `offer_type`. 
    * We will drop these columns, because for each column they are essentially all the same value. 
* The `num_photos` column looks strange.
We will need to look further into this. 

In [280]:
auto['seller'].value_counts()

privat        49999
gewerblich        1
Name: seller, dtype: int64

In [281]:
auto.loc[auto['seller']=='gewerblich']

date_crawled  ...            last_seen
7738  2016-03-15 18:06:22  ...  2016-04-06 17:15:37

[1 rows x 20 columns]

In [282]:
auto['offer_type'].value_counts()

Angebot    49999
Gesuch         1
Name: offer_type, dtype: int64

In [283]:
auto.loc[auto['offer_type']=='Gesuch']

date_crawled  ...            last_seen
17541  2016-04-03 15:48:33  ...  2016-04-05 15:16:06

[1 rows x 20 columns]

In [284]:
auto.loc[(auto['vehicle_type']=='bus') & (auto['brand']=='volkswagen'),'vehicle_type'].value_counts()

bus    1400
Name: vehicle_type, dtype: int64

>We discovered that there are only two different sellers and because the commercial seller is similar to the other data, we would be able to drop this column. This wouldn't have any affect to our analysis. 

>Also, we found out that there is only one Gesuch, compared to the Angebot's 49,999. After looking into the one Gesuch, we see the value is the same as the rest so we can keep the row, however we can drop the column without affecting the analysis.

In [285]:
auto['no_of_pictures'].value_counts()

0    50000
Name: no_of_pictures, dtype: int64

As we see, the values are 0 in this column allowing us to drop this column from the analysis.

In [286]:
auto.drop(columns=['seller','offer_type','no_of_pictures'], inplace=True)

###Analysis of Year of Registration

In this portion we will analyize the amount of cars that are registered in a particular time frame and remove the periods that have an insufficent amount of data that would otherwise act as outliers to our other analyses.


In [287]:
print(auto['reg_year'].value_counts(bins=5).sort_index())
auto['reg_year'].sort_values().unique()

(991.0, 2799.8]     49982
(2799.8, 4599.6]        3
(4599.6, 6399.4]        7
(6399.4, 8199.2]        0
(8199.2, 9999.0]        8
Name: reg_year, dtype: int64


array([1000, 1001, 1111, 1500, 1800, 1910, 1927, 1929, 1931, 1934, 1937,
       1938, 1939, 1941, 1943, 1948, 1950, 1951, 1952, 1953, 1954, 1955,
       1956, 1957, 1958, 1959, 1960, 1961, 1962, 1963, 1964, 1965, 1966,
       1967, 1968, 1969, 1970, 1971, 1972, 1973, 1974, 1975, 1976, 1977,
       1978, 1979, 1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988,
       1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
       2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
       2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019, 2800, 4100,
       4500, 4800, 5000, 5911, 6200, 8888, 9000, 9996, 9999])

After seeing the data, it's obvious that most the years are below 2799. Now we will go a little deeper into this data with some different facts. The first is we can take out the rows 1000 to 1800 because the first automobile was invented in 1885-1886. The second is that the data was collected in 2016, meaning dates past this can be wrong. So we will check out how many values there are after this year.



In [288]:
print("Number of Cars registered as between 1000 and 1885:", auto[auto['reg_year'].between(1000,1885)].shape[0])

print("Number of Cars registered as between 1885 and 1950: ", auto[auto['reg_year'].between(1885,1950)].shape[0])

print("Number of Cars registered as between 1950 and 1970: ", auto[auto['reg_year'].between(1950,1970)].shape[0])

print("Number of Cars registered as between 1970 and 1990: ", auto[auto['reg_year'].between(1970,1990)].shape[0])

print("Number of Cars registered as between 1990 and 2000: ", auto[auto['reg_year'].between(1990,2000)].shape[0])

print("Number of Cars registered as between 2000 and 2016: ", auto[auto['reg_year'].between(2000,2016)].shape[0])

print("Number of Cars registered as between 2016 and 2020: ", auto[auto['reg_year'].between(2016,2020)].shape[0])

print("Number of Cars registered as between 2020 and 9999: ", auto[auto['reg_year'].between(2020,9999)].shape[0])

print("Values within the range", auto[auto['reg_year'].between(1910,2016)].shape[0]/auto.shape[0]*100, "%")

Number of Cars registered as between 1000 and 1885: 6
Number of Cars registered as between 1885 and 1950:  27
Number of Cars registered as between 1950 and 1970:  250
Number of Cars registered as between 1970 and 1990:  1558
Number of Cars registered as between 1990 and 2000:  15839
Number of Cars registered as between 2000 and 2016:  34151
Number of Cars registered as between 2016 and 2020:  3264
Number of Cars registered as between 2020 and 9999:  18
Values within the range 96.056 %


We will now drop all the rows outside of 1910-2016 because of the lack of values from these other years.

In [289]:
auto['reg_month'].value_counts().sort_index()

0     5075
1     3282
2     3008
3     5071
4     4102
5     4107
6     4368
7     3949
8     3191
9     3389
10    3651
11    3360
12    3447
Name: reg_month, dtype: int64

In [290]:
auto.loc[auto['reg_month']==0].head()

date_crawled  ...            last_seen
11  2016-03-16 18:45:34  ...  2016-03-16 18:45:34
18  2016-03-26 19:57:44  ...  2016-04-02 12:16:41
27  2016-03-27 18:45:01  ...  2016-03-27 18:45:01
46  2016-03-31 10:53:28  ...  2016-04-06 03:44:41
52  2016-03-25 18:50:03  ...  2016-04-07 00:46:00

[5 rows x 17 columns]

After looking into the reg_month, we came to the conclusion that 0 might be a default option for users and since it has the most value we will not be dropping it.

In [291]:
auto = auto.loc[(auto['reg_year']>=1910)&(auto['reg_year']<=2016)]

###Analysis of Power PS

This section analyzes of the engine power (measured in PS) of the cars sold on Germany's Ebay.

In [292]:
auto['power_ps'].value_counts(bins=10).sort_index()

(-17.701, 1770.0]     48004
(1770.0, 3540.0]         11
(3540.0, 5310.0]          2
(5310.0, 7080.0]          4
(7080.0, 8850.0]          2
(8850.0, 10620.0]         0
(10620.0, 12390.0]        0
(12390.0, 14160.0]        1
(14160.0, 15930.0]        2
(15930.0, 17700.0]        2
Name: power_ps, dtype: int64

In [293]:
print(auto[auto['power_ps']>2500].shape[0])
auto[auto['power_ps']>2500].head()

15


date_crawled  ...            last_seen
1699   2016-04-04 19:49:19  ...  2016-04-06 22:16:46
3753   2016-04-03 18:47:14  ...  2016-04-05 19:17:39
10659  2016-03-18 11:47:40  ...  2016-04-03 21:17:20
11009  2016-03-15 21:55:11  ...  2016-04-07 07:45:21
11311  2016-04-01 02:36:43  ...  2016-04-07 05:17:11

[5 rows x 17 columns]

After looking into the power_ps, we see there are 17 entries that has 2500+ power value. We will leave these values alone for now, but we will come back to these higher values if there is a specific analysis needed for power.

In [294]:
auto.describe(include='all')

date_crawled         name  ...   postal_code            last_seen
count                 48028        48028  ...  48028.000000                48028
unique                46375        37019  ...           NaN                38102
top     2016-03-27 22:55:05  Ford_Fiesta  ...           NaN  2016-04-07 06:17:27
freq                      3           76  ...           NaN                    8
mean                    NaN          NaN  ...  50935.867327                  NaN
std                     NaN          NaN  ...  25792.079828                  NaN
min                     NaN          NaN  ...   1067.000000                  NaN
25%                     NaN          NaN  ...  30459.000000                  NaN
50%                     NaN          NaN  ...  49696.000000                  NaN
75%                     NaN          NaN  ...  71665.000000                  NaN
max                     NaN          NaN  ...  99998.000000                  NaN

[11 rows x 17 columns]

In [295]:
auto["price_$"].value_counts().head(25)

0.0       1335
500.0      757
1500.0     696
2500.0     614
1200.0     606
1000.0     602
600.0      511
3500.0     480
800.0      470
2000.0     438
999.0      413
750.0      402
650.0      394
850.0      388
4500.0     375
900.0      374
300.0      373
700.0      373
2200.0     361
1100.0     355
950.0      354
3000.0     352
1300.0     350
550.0      338
1800.0     335
Name: price_$, dtype: int64

* Once again, it appears the values in this column are rounded. 

* 1421 cars are listed as having a value of zero. We may want to delete delete these rows, they only compose a small amount of the dataset and may be considered outliers. 

* The highest price listed is one-million dollars, next we will look at the next highest prices in the dataset. 

In [296]:
print(round(auto["price_$"].describe() ,2))

count       48028.00
mean         9585.25
std        484381.73
min             0.00
25%          1150.00
50%          2990.00
75%          7400.00
max      99999999.00
Name: price_$, dtype: float64


## Exploring the Date Columns

There are five columns representing date values:

- `date_crawled`: added by the crawler
- `last_seen`: added by the crawler
- `ad_created`: from the website
- `registration_month`: from the website
- `registration_year`: from the website

The `date_crawled`, `last_seen`, and `ad_created` columns are currently stored as strings, we will need to convert these to numerical values. 

In [297]:
auto[auto['price_$'].between(250000,400000)].sort_values('price_$', ascending=False)

date_crawled  ...            last_seen
36818  2016-03-27 18:37:37  ...  2016-03-27 18:37:37
14715  2016-03-30 08:37:24  ...  2016-04-07 00:16:26
34723  2016-03-23 16:37:29  ...  2016-04-06 16:44:50
35923  2016-04-03 07:56:23  ...  2016-04-03 08:56:20
12682  2016-03-28 22:48:01  ...  2016-04-05 03:44:51
47337  2016-04-05 10:25:38  ...  2016-04-05 12:07:32
38299  2016-03-28 22:25:25  ...  2016-03-28 22:25:25

[7 rows x 17 columns]

We just went through and looked at the entries listed at 0(1335), meaning free cars. This is below 3% of the total entries so we will remove these values. Going foward, we will be looking at entries between the price range of 1-350,000 after removing expensive and free cars.

In [298]:
auto.describe()

price_$     reg_year  ...     reg_month   postal_code
count  4.802800e+04  48028.00000  ...  48028.000000  48028.000000
mean   9.585252e+03   2002.80351  ...      5.767760  50935.867327
std    4.843817e+05      7.31085  ...      3.696802  25792.079828
min    0.000000e+00   1910.00000  ...      0.000000   1067.000000
25%    1.150000e+03   1999.00000  ...      3.000000  30459.000000
50%    2.990000e+03   2003.00000  ...      6.000000  49696.000000
75%    7.400000e+03   2008.00000  ...      9.000000  71665.000000
max    1.000000e+08   2016.00000  ...     12.000000  99998.000000

[8 rows x 6 columns]

## Exploring the `odometer` and `price_in_dollars` Columns

In [299]:
auto["odometer_km"].value_counts().head(25)

150000.0    31029
125000.0     4960
100000.0     2110
90000.0      1696
80000.0      1396
70000.0      1199
60000.0      1137
50000.0      1008
5000.0        911
40000.0       801
30000.0       769
20000.0       763
10000.0       249
Name: odometer_km, dtype: int64

* We found that in the `odometer` column, there are many more high-millage vehicles being sold compared to low-millage vehicles. 

* Also, it appears the odometer values have been rounded.

In [300]:
print(auto['odometer_km'].describe())
auto['odometer_km'].value_counts()

count     48028.000000
mean     125544.161739
std       40106.751417
min        5000.000000
25%      100000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64


150000.0    31029
125000.0     4960
100000.0     2110
90000.0      1696
80000.0      1396
70000.0      1199
60000.0      1137
50000.0      1008
5000.0        911
40000.0       801
30000.0       769
20000.0       763
10000.0       249
Name: odometer_km, dtype: int64

## Exploring the Date Columns

There are five columns representing date values:

- `date_crawled`: added by the crawler
- `last_seen`: added by the crawler
- `ad_created`: from the website
- `registration_month`: from the website
- `registration_year`: from the website

The `date_crawled`, `last_seen`, and `ad_created` columns are currently stored as strings, we will need to convert these to numerical values. **bold text**

In [301]:
auto[['date_crawled','date_created','last_seen']].head()

date_crawled         date_created            last_seen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [302]:
print(round(auto["date_crawled"].str[:10].value_counts(normalize=True, dropna=False).sort_index() ,2))

2016-03-05    0.03
2016-03-06    0.01
2016-03-07    0.04
2016-03-08    0.03
2016-03-09    0.03
2016-03-10    0.03
2016-03-11    0.03
2016-03-12    0.04
2016-03-13    0.02
2016-03-14    0.04
2016-03-15    0.03
2016-03-16    0.03
2016-03-17    0.03
2016-03-18    0.01
2016-03-19    0.03
2016-03-20    0.04
2016-03-21    0.04
2016-03-22    0.03
2016-03-23    0.03
2016-03-24    0.03
2016-03-25    0.03
2016-03-26    0.03
2016-03-27    0.03
2016-03-28    0.03
2016-03-29    0.03
2016-03-30    0.03
2016-03-31    0.03
2016-04-01    0.03
2016-04-02    0.04
2016-04-03    0.04
2016-04-04    0.04
2016-04-05    0.01
2016-04-06    0.00
2016-04-07    0.00
Name: date_crawled, dtype: float64


In [303]:
print(round(auto["date_created"].str[:10].value_counts(normalize=True, dropna=False).sort_index() ,4)*100)

2015-06-11    0.00
2015-08-10    0.00
2015-09-09    0.00
2015-11-10    0.00
2015-12-05    0.00
              ... 
2016-04-03    3.91
2016-04-04    3.70
2016-04-05    1.18
2016-04-06    0.32
2016-04-07    0.13
Name: date_created, Length: 74, dtype: float64


In [304]:
print(round(auto["last_seen"].str[:10].value_counts(normalize=True, dropna=False).sort_index() ,4)*100)

2016-03-05     0.11
2016-03-06     0.42
2016-03-07     0.54
2016-03-08     0.77
2016-03-09     1.00
2016-03-10     1.08
2016-03-11     1.25
2016-03-12     2.38
2016-03-13     0.88
2016-03-14     1.28
2016-03-15     1.60
2016-03-16     1.62
2016-03-17     2.79
2016-03-18     0.73
2016-03-19     1.56
2016-03-20     2.06
2016-03-21     2.07
2016-03-22     2.11
2016-03-23     1.84
2016-03-24     1.95
2016-03-25     1.89
2016-03-26     1.69
2016-03-27     1.60
2016-03-28     2.07
2016-03-29     2.21
2016-03-30     2.47
2016-03-31     2.37
2016-04-01     2.33
2016-04-02     2.47
2016-04-03     2.53
2016-04-04     2.42
2016-04-05    12.49
2016-04-06    22.24
2016-04-07    13.18
Name: last_seen, dtype: float64


After checking both the odometer and date columns, most the data seems to be normal. So, we will not be changing anything with both of the columns. 

Now, we will look to see if there are any correlations between differents categories like brands price, distance, and age of the vehicle. 

#Analyzing Sale Patterns on Ebay Germany

We will look at the vehicle types sold and what their portion of the Ebay market is, and the percentage of cars that are for sale that have damage vs those that do not. Our analysis continues with the creation of a variable that represents the age and the length of time it took to sell the vehicles. We will find the average of these variables as well as the average milage and price with respect to the top brands and, more specifically, the top models. 

##We begin by looking at the vehicle type and its portion of the market

In [305]:
type_popularity=round(auto["vehicle_type"].value_counts(normalize=True) ,4)*100
print(type_popularity)

limousine     28.64
kleinwagen    24.10
kombi         20.32
bus            9.12
cabrio         6.82
coupe          5.65
suv            4.42
andere         0.94
Name: vehicle_type, dtype: float64


###########write something

#Percentage of Cars being sold with damage

In [306]:
sold_with_damage=round(auto["not_repaired_damage"].value_counts(normalize=True) ,4)*100
print(sold_with_damage)

nein    87.74
ja      12.26
Name: not_repaired_damage, dtype: float64


######write something :)

#Analyis of Top Brands and Models

In [307]:
auto['age']=2016-auto['reg_year']
auto.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


date_crawled  ... age
0  2016-03-26 17:47:46  ...  12
1  2016-04-04 13:38:56  ...  19
2  2016-03-26 18:57:24  ...   7
3  2016-03-12 16:58:10  ...   9
4  2016-04-01 14:38:50  ...  13

[5 rows x 18 columns]

In [308]:
crawled = pd.to_datetime(auto['date_crawled'].str[:10])
posted = pd.to_datetime(auto['date_created'].str[:10])
sold = pd.to_datetime(auto['last_seen'].str[:10])

time_to_sell = sold - posted 
auto['time_to_sell'] = time_to_sell

auto['time_to_sell'].value_counts(bins=10).sort_index()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


(-1 days +16:47:59.999999, 30 days 00:00:00]    47300
(30 days 00:00:00, 60 days 00:00:00]              716
(60 days 00:00:00, 90 days 00:00:00]                5
(90 days 00:00:00, 120 days 00:00:00]               3
(120 days 00:00:00, 150 days 00:00:00]              1
(150 days 00:00:00, 180 days 00:00:00]              0
(180 days 00:00:00, 210 days 00:00:00]              1
(210 days 00:00:00, 240 days 00:00:00]              1
(240 days 00:00:00, 270 days 00:00:00]              0
(270 days 00:00:00, 300 days 00:00:00]              1
Name: time_to_sell, dtype: int64

As you see, most the users had success within the first 30 days after listing the car. However, there are a few entries that have gone many days past this range. It is possible that the age of the car or the price for the car is what is factoring into these outliers. 

In [309]:
auto['time_to_sell_int']=auto['time_to_sell'].astype(str).str.split().str[0].astype(int)


auto.head()

/usr/local/lib/python3.6/dist-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


date_crawled  ... time_to_sell_int
0  2016-03-26 17:47:46  ...               11
1  2016-04-04 13:38:56  ...                2
2  2016-03-26 18:57:24  ...               11
3  2016-03-12 16:58:10  ...                3
4  2016-04-01 14:38:50  ...                0

[5 rows x 20 columns]

In [310]:
auto['brand'].value_counts(normalize=True)

volkswagen        0.212126
bmw               0.110019
opel              0.108166
mercedes_benz     0.095361
audi              0.086387
ford              0.069793
renault           0.047347
peugeot           0.029524
fiat              0.025860
seat              0.018177
skoda             0.016032
mazda             0.015137
nissan            0.015095
citroen           0.013929
smart             0.013909
toyota            0.012472
sonstige_autos    0.010952
hyundai           0.009848
volvo             0.009245
mini              0.008641
mitsubishi        0.008141
honda             0.007850
kia               0.007100
alfa_romeo        0.006621
porsche           0.006101
suzuki            0.005913
chevrolet         0.005705
chrysler          0.003665
dacia             0.002561
daihatsu          0.002561
jeep              0.002249
subaru            0.002186
land_rover        0.002040
saab              0.001603
jaguar            0.001582
trabant           0.001562
daewoo            0.001499
r

From the results, we see that the top 5 brands are: Volkswagon, BMW, Opel, Mercedes Benz, and Audi. Now, we want to look into the mean values from some columns and seee if they correlate to these different brands. 

In [311]:
brand_price_dict={}
brand_distance_dict={}
brand_age_dict={}
brand_sell_time_dict={}

brand_names= auto['brand'].unique()

for brand_name in brand_names:
  mean_price= auto.loc[auto['brand']==brand_name,'price_$'].mean() 
  mean_distance=auto.loc[auto['brand']==brand_name,'odometer_km'].mean()
  mean_age=auto.loc[auto['brand']==brand_name,'age'].mean()
  mean_sell_time =auto.loc[auto['brand']==brand_name,'time_to_sell_int'].mean()

  brand_price_dict[brand_name] = int(mean_price)
  brand_distance_dict[brand_name] = int(mean_distance)
  brand_age_dict[brand_name] = int(mean_age)
  brand_sell_time_dict[brand_name] = int(mean_sell_time)

In [312]:
brand_mean_price_series = pd.Series(brand_price_dict)
brand_mean_distance_series = pd.Series(brand_distance_dict)
brand_mean_age_series = pd.Series(brand_age_dict)
brand_mean_sell_time_series=pd.Series(brand_sell_time_dict)

brand_data = pd.DataFrame(brand_mean_price_series,columns=['mean_price'])

brand_data['mean_distance'] = brand_mean_distance_series
brand_data['mean_age'] = brand_mean_age_series
brand_data['mean_sell_time']=brand_mean_sell_time_series
brand_data['total_percentage']=auto['brand'].value_counts(normalize=True)*100


brand_data.head(10).sort_values('total_percentage', ascending=False)

mean_price  mean_distance  ...  mean_sell_time  total_percentage
volkswagen           6516         128730  ...               8         21.212626
bmw                  8334         132434  ...               9         11.001916
mercedes_benz       30317         130860  ...               9          9.536104
audi                 9093         129287  ...               9          8.638711
ford                 7263         124046  ...               8          6.979262
renault              2395         128183  ...               8          4.734738
peugeot              3039         127136  ...               8          2.952444
seat                 4296         121563  ...               8          1.817690
smart                3542          99595  ...               9          1.390855
chrysler             3229         133181  ...               9          0.366453

[10 rows x 5 columns]

The mean values for the coulumns are relatively the same. However, when it comes to the number of listings, Volkswagon almost doubles the second brand. A reason for this could come from the fact that the price is actually moderate compared to the other higher listed brands. 

In [313]:
print(brand_data['mean_price'][['mercedes_benz','peugeot','opel','subaru','smart']])

auto[auto['time_to_sell_int']>60].sort_values(
    'time_to_sell_int',ascending=False)[['brand','age','price_$','odometer_km','time_to_sell']]

mercedes_benz    30317
peugeot           3039
opel              5252
subaru            3765
smart             3542
Name: mean_price, dtype: int64


brand  age  price_$  odometer_km time_to_sell
22781  mercedes_benz    2  47900.0      20000.0     300 days
20649        peugeot   10   5200.0      80000.0     239 days
34883           opel    3   9500.0      50000.0     209 days
2243         peugeot    7   5499.0      70000.0     149 days
27986         subaru   19    900.0     150000.0     109 days
36993          smart   12   2400.0      90000.0      98 days
2232      volkswagen   10  14499.0      90000.0      95 days
24851  mercedes_benz    8  15490.0     150000.0      88 days
39270           opel   19   1099.0     150000.0      68 days
15081           audi   11   5800.0     150000.0      63 days
49743            bmw    9   7399.0     100000.0      62 days
36429  mercedes_benz   11   9900.0     150000.0      61 days

We wanted to look into the cars that had trouble getting sold. After seeing the data, it is obvious that the price is the main reason for why the longest entry had so much trouble. There is another entry in which the car is just so old that most people would not want to buy it. Overall, there are a few reasons for why some cars just don't get sold; these would be that the car could be too much money compared to the mean price, or that some cars are too old and people would wanted to get cars with less years on them.

In [314]:
print(auto['time_to_sell'].value_counts(bins=10).sort_index())

print(brand_data['mean_price'][['jeep','ford','volkswagen','audi','bmw','smart']])

auto.sort_values('time_to_sell_int')[
    ['brand','age','price_$','odometer_km','time_to_sell','date_created','last_seen','date_crawled']].head(20)


(-1 days +16:47:59.999999, 30 days 00:00:00]    47300
(30 days 00:00:00, 60 days 00:00:00]              716
(60 days 00:00:00, 90 days 00:00:00]                5
(90 days 00:00:00, 120 days 00:00:00]               3
(120 days 00:00:00, 150 days 00:00:00]              1
(150 days 00:00:00, 180 days 00:00:00]              0
(180 days 00:00:00, 210 days 00:00:00]              1
(210 days 00:00:00, 240 days 00:00:00]              1
(240 days 00:00:00, 270 days 00:00:00]              0
(270 days 00:00:00, 300 days 00:00:00]              1
Name: time_to_sell, dtype: int64
jeep          11434
ford           7263
volkswagen     6516
audi           9093
bmw            8334
smart          3542
Name: mean_price, dtype: int64


brand  age  ...            last_seen         date_crawled
44848  volkswagen   20  ...  2016-03-29 07:38:17  2016-03-29 01:56:53
5190       nissan   19  ...  2016-04-02 11:56:06  2016-04-02 11:56:06
25518        audi   10  ...  2016-04-02 10:47:09  2016-04-02 09:36:41
25519       skoda    9  ...  2016-04-01 13:37:59  2016-04-01 13:37:59
9879      peugeot   15  ...  2016-04-01 14:48:54  2016-04-01 14:48:54
18142       skoda    9  ...  2016-04-01 10:40:34  2016-04-01 09:51:08
18141        ford   15  ...  2016-03-17 19:47:38  2016-03-17 19:47:38
37576        ford   19  ...  2016-04-02 19:41:06  2016-04-02 18:53:21
5182         opel   10  ...  2016-03-08 16:52:55  2016-03-08 16:52:55
47518       smart   12  ...  2016-03-15 14:47:44  2016-03-15 14:47:44
25533  volkswagen    0  ...  2016-03-29 19:42:08  2016-03-29 18:58:54
18139        opel    0  ...  2016-03-17 17:39:53  2016-03-17 16:56:16
18146        opel   18  ...  2016-03-22 14:49:44  2016-03-22 14:49:44
25537     citroen    1  ...  2016-03-16 15:41:39  2016-03-16 14:52:45
45178  volkswagen    0  ...  2016-04-01 11:53:32  2016-04-01 11:53:32
25542     peugeot    6  ...  2016-03-23 17:06:47  2016-03-23 16:38:19
45176     citroen   11  ...  2016-03-21 19:50:51  2016-03-21 19:50:51
5170         ford    4  ...  2016-04-02 19:57:57  2016-04-02 19:57:57
9889         fiat   12  ...  2016-03-11 22:15:57  2016-03-11 06:56:17
18127        opel   16  ...  2016-03-17 19:53:55  2016-03-17 19:53:55

[20 rows x 8 columns]

In [315]:
print(auto['time_to_sell_int'].value_counts().head(10).sort_index(), '\n')

for i in range(20):
  name_of_brand=auto.sort_values('time_to_sell_int').iloc[i]['brand']  
  list_price = auto.sort_values('time_to_sell_int').iloc[i]['price_$']  
  mean_price = brand_data.loc[name_of_brand]['mean_price']             
  difference = mean_price-list_price                                    
  print(difference, ":", name_of_brand)

0    6728
1    2778
2    5003
3    2417
4    3574
5    1622
6    2661
7    1652
8    2284
9    1759
Name: time_to_sell_int, dtype: int64 

6217.0 : volkswagen
3965.0 : nissan
-2897.0 : audi
934.0 : skoda
1339.0 : peugeot
1735.0 : skoda
6863.0 : ford
6273.0 : ford
5252.0 : opel
2442.0 : smart
5866.0 : volkswagen
2253.0 : opel
4402.0 : opel
36784.0 : citroen
5666.0 : volkswagen
2489.0 : peugeot
40834.0 : citroen
-2727.0 : ford
1511.0 : fiat
4102.0 : opel


In [316]:
auto1=auto[auto['time_to_sell_int']==1].sort_values('time_to_sell')

for i in range(20):
  name_of_brand=auto1.sort_values('time_to_sell_int').iloc[i]['brand']   
  list_price = auto1.sort_values('time_to_sell_int').iloc[i]['price_$']  
  mean_price = brand_data.loc[name_of_brand]['mean_price']              
  difference = mean_price-list_price                                    
  print(difference, ":", name_of_brand)

auto1.sort_values('time_to_sell')[
    ['brand','age','price_$','odometer_km','time_to_sell','date_created','last_seen','date_crawled']].head(20)


20721.0 : sonstige_autos
2411.0 : fiat
6066.0 : volkswagen
1245.0 : renault
2215.0 : toyota
1229.0 : chrysler
4252.0 : opel
-3516.0 : alfa_romeo
302.0 : opel
25117.0 : mercedes_benz
4662.0 : opel
6813.0 : ford
2634.0 : mitsubishi
-3734.0 : volkswagen
4226.0 : volkswagen
7263.0 : ford
6516.0 : volkswagen
29067.0 : mercedes_benz
6326.0 : volkswagen
23617.0 : mercedes_benz


brand  age  ...            last_seen         date_crawled
15   sonstige_autos   34  ...  2016-04-02 21:10:48  2016-04-01 12:06:20
174            fiat   16  ...  2016-03-20 15:47:12  2016-03-19 13:53:19
176      volkswagen   19  ...  2016-03-14 13:28:22  2016-03-13 16:22:15
199         renault   17  ...  2016-03-13 00:45:34  2016-03-12 08:54:25
225          toyota   19  ...  2016-03-16 05:18:46  2016-03-15 09:57:37
275        chrysler   15  ...  2016-03-08 10:44:31  2016-03-07 18:41:56
295            opel    0  ...  2016-03-29 23:18:11  2016-03-28 03:36:22
783      alfa_romeo   10  ...  2016-03-17 16:47:47  2016-03-16 17:57:22
333            opel   11  ...  2016-03-24 07:40:54  2016-03-23 23:37:23
363   mercedes_benz   10  ...  2016-04-05 12:22:58  2016-04-04 18:52:13
385            opel   18  ...  2016-03-23 22:45:32  2016-03-22 14:56:32
402            ford   11  ...  2016-03-06 03:45:35  2016-03-05 21:49:41
408      mitsubishi   16  ...  2016-03-16 20:21:44  2016-03-15 21:49:23
411      volkswagen    7  ...  2016-03-18 00:41:13  2016-03-17 22:38:14
413      volkswagen   11  ...  2016-03-22 07:42:14  2016-03-22 00:51:35
430            ford    9  ...  2016-03-19 06:47:06  2016-03-18 23:52:40
449      volkswagen   15  ...  2016-04-04 10:31:06  2016-04-03 20:59:59
334   mercedes_benz   16  ...  2016-03-15 04:45:49  2016-03-14 09:51:24
127      volkswagen   21  ...  2016-03-22 20:18:58  2016-03-21 15:49:37
123   mercedes_benz   13  ...  2016-03-22 20:16:50  2016-03-21 15:46:24

[20 rows x 8 columns]

In [317]:
auto2=auto[auto['time_to_sell_int']==9].sort_values('time_to_sell')

for i in range(20):
  name_of_brand=auto2.sort_values('time_to_sell_int').iloc[i]['brand']   
  list_price = auto2.sort_values('time_to_sell_int').iloc[i]['price_$']  
  mean_price = brand_data.loc[name_of_brand]['mean_price']              
  difference = mean_price-list_price                                   
  print(difference, ":", name_of_brand)

auto2.sort_values('time_to_sell')[
    ['brand','age','price_$','odometer_km','time_to_sell','date_created','last_seen','date_crawled']].head(20)

7743.0 : audi
-1604.0 : renault
-25683.0 : mercedes_benz
-2189.0 : fiat
4666.0 : volkswagen
3652.0 : opel
-5453.0 : smart
15817.0 : mercedes_benz
27318.0 : mercedes_benz
1834.0 : bmw
6143.0 : audi
-3454.0 : seat
24317.0 : mercedes_benz
-7235.0 : mini
2402.0 : opel
4502.0 : opel
6234.0 : bmw
-9826.0 : nissan
3816.0 : volkswagen
-3484.0 : volkswagen


brand  age  ...            last_seen         date_crawled
13            audi   17  ...  2016-04-01 14:17:13  2016-03-23 10:48:32
14         renault    9  ...  2016-04-01 15:46:47  2016-03-23 11:50:46
49   mercedes_benz    4  ...  2016-04-05 11:47:52  2016-03-27 15:36:42
162           fiat    7  ...  2016-04-06 13:44:51  2016-03-28 10:54:23
204     volkswagen   16  ...  2016-03-19 18:17:21  2016-03-10 23:45:19
222           opel   20  ...  2016-04-05 11:47:51  2016-03-27 15:58:44
267          smart    3  ...  2016-04-06 13:17:27  2016-03-28 11:37:34
289  mercedes_benz   20  ...  2016-04-05 20:46:01  2016-03-27 20:53:07
298  mercedes_benz   18  ...  2016-04-06 22:17:48  2016-03-28 16:54:57
319            bmw   11  ...  2016-03-24 23:46:20  2016-03-15 09:50:32
326           audi   16  ...  2016-03-16 00:44:31  2016-03-08 19:41:03
356           seat    4  ...  2016-03-17 17:47:27  2016-03-08 10:51:56
373  mercedes_benz   13  ...  2016-03-31 17:15:37  2016-03-22 18:42:27
403           mini    1  ...  2016-04-06 13:44:50  2016-03-28 10:53:59
425           opel   11  ...  2016-04-02 22:19:37  2016-03-24 09:46:21
445           opel   20  ...  2016-03-17 20:21:50  2016-03-08 17:57:42
471            bmw   17  ...  2016-03-19 15:46:03  2016-03-10 10:51:21
505         nissan    4  ...  2016-03-14 12:45:10  2016-03-05 20:57:20
569     volkswagen   15  ...  2016-04-05 12:21:24  2016-03-27 16:48:21
574     volkswagen    0  ...  2016-04-05 14:47:50  2016-03-27 18:39:33

[20 rows x 8 columns]

Differnt observations from fast selling cars:


*  The lower the price is compared to the mean value of a brand, the faster the car will be sold.
*   Age of the car also is important in how fast a car is sold. An example is that if a car is new, the price will most likely be higher compared to the mean price, thus the listing will take longer to be sold.



In [318]:
auto['age'].value_counts(bins=10).sort_index()

(-0.107, 10.6]    17082
(10.6, 21.2]      27307
(21.2, 31.8]       2826
(31.8, 42.4]        451
(42.4, 53.0]        265
(53.0, 63.6]         67
(63.6, 74.2]          8
(74.2, 84.8]         10
(84.8, 95.4]          3
(95.4, 106.0]         9
Name: age, dtype: int64

In [319]:
auto[auto['age']>84].sort_values('price_$')[['brand','age','price_$','odometer_km','time_to_sell','date_created','last_seen']]

brand  age  ...         date_created            last_seen
15898  sonstige_autos  106  ...  2016-03-08 00:00:00  2016-03-08 17:47:19
33295  sonstige_autos  106  ...  2016-03-12 00:00:00  2016-03-19 07:15:34
42181  sonstige_autos  106  ...  2016-03-27 00:00:00  2016-04-06 01:15:30
46213  sonstige_autos  106  ...  2016-04-02 00:00:00  2016-04-04 11:16:30
3679   sonstige_autos  106  ...  2016-04-04 00:00:00  2016-04-04 07:49:15
45157         trabant  106  ...  2016-03-11 00:00:00  2016-03-25 11:18:57
30781            opel  106  ...  2016-03-25 00:00:00  2016-03-26 23:46:29
22659            opel  106  ...  2016-03-14 00:00:00  2016-04-03 07:53:55
28693         renault  106  ...  2016-03-22 00:00:00  2016-04-06 09:16:59
22101             bmw   87  ...  2016-03-09 00:00:00  2016-04-07 06:17:11
21416            ford   89  ...  2016-03-12 00:00:00  2016-03-15 12:45:12
11246            ford   85  ...  2016-03-26 00:00:00  2016-04-06 09:46:59

[12 rows x 7 columns]

Checking out the extremely old cars, it would most likely be like incorrect data because vintage cars tend to cost a lot more. So, we can take these out of the analysis.

In [320]:
auto[auto['age']<5].sort_values('price_$').head(10)[['brand','age','price_$','odometer_km','time_to_sell']]

brand  age  price_$  odometer_km time_to_sell
17743        audi    0      0.0     150000.0       4 days
34291        ford    0      0.0       5000.0      31 days
37816        audi    0      0.0      40000.0       0 days
40667         bmw    0      0.0     150000.0      10 days
19538        seat    1      0.0      10000.0       0 days
24871  volkswagen    0      0.0     150000.0       2 days
4079         opel    0      0.0       5000.0       1 days
46752        opel    0      0.0      90000.0       4 days
7477   volkswagen    0      0.0     150000.0       7 days
10755  volkswagen    0      0.0     150000.0       0 days

Most likely, this is also unreliable data because new cars would typically not have 150,000 km.

In [321]:
auto[auto['age']<5].sort_values('price_$', ascending=False).head(10)[['brand','age','price_$','odometer_km','time_to_sell']]

brand  age     price_$  odometer_km time_to_sell
42221         citroen    2  27322222.0      40000.0       0 days
36818         porsche    0    350000.0       5000.0       0 days
14715  sonstige_autos    4    345000.0      20000.0       8 days
35923         porsche    1    295000.0       5000.0       0 days
12682         porsche    0    265000.0       5000.0       8 days
38299             bmw    1    250000.0       5000.0       0 days
40918         porsche    1    198000.0       5000.0       1 days
28090  sonstige_autos    0    194000.0       5000.0       8 days
17140   mercedes_benz    4    180000.0      50000.0      15 days
11433            audi    0    175000.0       5000.0       6 days

In [322]:
brand_data[['mean_price', 'mean_age','mean_distance']].loc[['porsche','bmw','mercedes_benz']]

mean_price  mean_age  mean_distance
porsche             44553        14          97457
bmw                  8334        13         132434
mercedes_benz       30317        14         130860

Even though the price might be higher than the mean price for a brand, because it is new compared to the mean age of the brand the car will still be high in demand.

In [323]:
auto[auto['age'].between(5,10)].sort_values('price_$').head(10)[['brand','age','price_$','odometer_km','time_to_sell']]

brand  age  price_$  odometer_km time_to_sell
16973  renault   10      0.0     150000.0       7 days
2207   renault   10      0.0     150000.0       3 days
3521     skoda    7      0.0     125000.0      14 days
15669      bmw    5      0.0      50000.0       0 days
45663      bmw    9      0.0     150000.0       7 days
45658     opel    9      0.0     150000.0       2 days
26922  renault    8      0.0      80000.0       1 days
43131      bmw   10      0.0     150000.0       1 days
849    renault    5      0.0      60000.0       1 days
23599      bmw    9      0.0       5000.0      11 days

In [324]:
auto[auto['age'].between(5,10)].sort_values('price_$', ascending=False).head(10)[
    ['brand','age','price_$','odometer_km','time_to_sell']]

brand  age    price_$  odometer_km time_to_sell
47634  sonstige_autos   10  3890000.0       5000.0       1 days
514              ford    7   999999.0     125000.0      20 days
37840         porsche    8   220000.0      30000.0      16 days
33638         porsche    9   151990.0      20000.0      14 days
32185         porsche    6   139997.0      20000.0      12 days
2454          porsche    6   137999.0      20000.0      17 days
14268         porsche    6   135000.0      30000.0       6 days
49391  sonstige_autos    8   109999.0      30000.0      19 days
16964  sonstige_autos    6   105000.0      80000.0       4 days
29286         porsche    5   104900.0      30000.0      15 days

In [325]:
popular_models=round(auto["model"].value_counts(normalize=True) ,4)*100
print(popular_models)

golf          8.37
andere        7.59
3er           5.90
polo          3.68
corsa         3.61
              ... 
kappa         0.00
200           0.00
i3            0.00
b_max         0.00
rangerover    0.00
Name: model, Length: 244, dtype: float64


In [326]:
top_popular_models = popular_models[popular_models > 3.0].index
print(top_popular_models)

Index(['golf', 'andere', '3er', 'polo', 'corsa', 'passat', 'astra'], dtype='object')


In [327]:
mean_mileage_by_model = {}

for model in top_popular_models:
    model_only_2 = auto[auto["model"] == model]
    mean_mileage_1 = model_only_2["odometer_km"].mean()
    mean_mileage_by_model[model] = round(mean_mileage_1 ,2)

mean_mileage_by_model

{'3er': 137189.36,
 'andere': 119943.59,
 'astra': 131095.1,
 'corsa': 128361.7,
 'golf': 128167.76,
 'passat': 139884.73,
 'polo': 127391.17}

In [329]:
model_average_price = {}

for model in top_popular_models:
    model_only = auto[auto["model"] == model]
    mean_price_for_model = model_only["price_$"].mean()
    model_average_price[model] = round(mean_price_for_model ,2)
    
model_average_price

{'3er': 5838.8,
 'andere': 6772.68,
 'astra': 3217.01,
 'corsa': 1816.05,
 'golf': 4937.72,
 'passat': 4809.26,
 'polo': 2488.23}

# **Conclusion**
After cleaning and analysing all of the datatset, we have come up with different conclusions from the date:


*   The top brands are Volkswagon, BMW, Opel, Mercedes Benz, and Audi.

*   Volkswagon is the most listed brand, probably because of its mean pricing compared to the other top brands.

*   Cars are difficult to sell if the pricing is higher than the mean price and if the car is older than the mean age.
*   To sell a car fast, have the price for the listing about $2000 lower than the mean value.


*   The newer a car is, the higher the price will be compared to the mean vaule, thus taking a longer time to sell.


*   Some cars can still be sold at older ages because of the antiquity and the brand the car is. These cars might have a higher price than the mean value, but it can still be sold relatively fast.

*   A car can still have high demand even when the price is higher than the mean value if; it is new has brand value, and has decent mileage. 




